File: app/src/main/java/ua/ck/zabochen/englishverbs/view/main/MainActivity.java
package ua.ck.zabochen.englishverbs.view.main;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.FrameLayout;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import ua.ck.zabochen.englishverbs.MainApp;
import ua.ck.zabochen.englishverbs.R;
import ua.ck.zabochen.englishverbs.view.base.BaseActivity;
import ua.ck.zabochen.englishverbs.view.setting.SettingActivity;
import ua.ck.zabochen.englishverbs.view.verb.VerbFragment;

public class MainActivity extends BaseActivity {

    @Inject
    VerbFragment mVerbFragment;

    @BindView(R.id.activityMain_toolbar)
    Toolbar mToolbar;

    @BindView(R.id.activityMain_frameLayout)
    FrameLayout mFrameLayout;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        MainApp.getAppComponent().inject(this);
        setUi();
        setFragment();
    }

    private void setUi() {
        // Layout
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        // Toolbar
        setSupportActionBar(mToolbar);
        mToolbar.setTitleTextColor(ContextCompat.getColor(this, R.color.toolbar_title));
    }

    private void setFragment() {
        getSupportFragmentManager().beginTransaction()
                .replace(mFrameLayout.getId(), mVerbFragment)
                .commit();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.menuMain_itemSettings:
                startActivity(new Intent(this, SettingActivity.class));
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/java/ua/ck/zabochen/englishverbs/view/main/MainActivity.kt
package ua.ck.zabochen.englishverbs.view.main

import android.os.Bundle
import android.support.design.widget.BottomNavigationView
import android.support.design.widget.CoordinatorLayout
import android.support.v4.app.Fragment
import android.support.v7.widget.Toolbar
import android.widget.FrameLayout
import com.hannesdorfmann.mosby3.mvp.MvpActivity
import org.jetbrains.anko.AnkoLogger
import ua.ck.zabochen.englishverbs.R
import ua.ck.zabochen.englishverbs.utils.behavior.BottomNavigationViewBehavior
import ua.ck.zabochen.englishverbs.view.bookmark.BookmarkFragment
import ua.ck.zabochen.englishverbs.view.notification.NotificationFragment
import ua.ck.zabochen.englishverbs.view.verblist.VerbListFragment

class MainActivity : MvpActivity<MainView, MainPresenter>(),
        MainView, AnkoLogger {

    private val mToolbar: Toolbar by lazy { findViewById<Toolbar>(R.id.snippet_toolbar) }
    private val mFrameLayout: FrameLayout by lazy { findViewById<FrameLayout>(R.id.activityMain_frameLayout) }
    private val mBottomNavigationView: BottomNavigationView by lazy { findViewById<BottomNavigationView>(R.id.activityMain_bottomNavigationView) }

    override fun createPresenter() = MainPresenter()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setUi()
        inflateOrUpdateDatabase()
    }

    private fun inflateOrUpdateDatabase() {
        presenter.inflateOrUpdateDatabase()
    }

    private fun setUi() {
        // Layout
        setContentView(R.layout.activity_main)

        // Toolbar
        setSupportActionBar(mToolbar)
        supportActionBar?.setDisplayShowTitleEnabled(false)

        // Bottom Navigation View - Behavior (Hide/Show)
        val layoutParams: CoordinatorLayout.LayoutParams = mBottomNavigationView.layoutParams as CoordinatorLayout.LayoutParams
        layoutParams.behavior = BottomNavigationViewBehavior()

        // Bottom Navigation View - Listener (Item selected)
        mBottomNavigationView.setOnNavigationItemSelectedListener { item ->
            when (item.itemId) {
                R.id.menuBottomNavigationView_item_home -> setFragment(VerbListFragment())
                R.id.menuBottomNavigationView_item_bookmarks -> setFragment(BookmarkFragment())
                R.id.menuBottomNavigationView_item_notifications -> setFragment(NotificationFragment())
            }
            return@setOnNavigationItemSelectedListener true
        }
    }

    override fun setFragment(fragment: Fragment) {
        supportFragmentManager.beginTransaction()
                .replace(mFrameLayout.id, fragment)
                .commit()
    }

}